js/sidebar.js

```diff
@@ -3,4 +3,13 @@ function openSideBar() {
 }
 function closeSideBar() {
 	document.getElementById("slide-out").style.transform = "translateX(-100%)";
-}
\ No newline at end of file
+}
+$(document).ready(function(){
+	$.ajax({
+    	url: "https://api.github.com/repos/fossasia/labyrinth/git/refs/heads/master"
```

Comment:
Does this work when the file is stored localy? When does this work? When does it not work? What about branches and pull requests?

Reply:
It gets the top most commit of the fossasia:master using the Github API. (It is a very effecient way as it doesn't get all the commits and then just takes the first one. It only takes the top one.)
If the json data is in a var called `data`
It then outputs `data.object.sha` which has its full sha256 id.

It will not work if commits are made locally or in any other branch other than the fossasia:master

I can try to get the sha of the latest commit from the `.git` folder but it is hidden by default and will be very complicated and inefficient. I will try it nevertheless.